[REJECTED] arm/include/math.h: remove _PARAMS macro

bkueng opened the pull request:
_PARAMS got removed in upstream newlib:
https://github.com/bminor/newlib/commit/eea249da3bc81776246ad5163f5eb887afdd3659

I marked this as [REJECTED] because it is based on a rejected commit (c44095ffacf4bc470d93a22fde601763685668d5) and the file is not present in upstream.

nuttx commented:
Committed upstream: http://sourceforge.net/p/nuttx/git/ci/1923fed668f74aafc8edcb13cb9cab07e50ba97e/

Some notes:

1. You version of stm32_rtc.c is not the same as the upstream GIT.  As a result the patch did not apply and parts of it had to be included manually.  You might want to review.  The basic problem is that the upstream version includes a retry loop and the PX4 version does not.  Changes were made to logic that is now within a loop.

2. The HSE must be divided down to get the 1MHz RTC clock input.  That divider was hard-coded to 24 for a 24MHz HSE clock.  Most STM32s do not have a 24MHz HSE clock.  I added logic to calculate the divider to get a 1MHz clock based on STM32_HSE_FREQUENCY which should be provided in every STM32 board.h header file.  This change is untested.

3. Various stylistic changes to conform with the NuttX coding style.

4. Also, there were no missing changes to the Kconfig file that would allow selecting the LSE/HSE RTC clock source options.  I added these:  http://sourceforge.net/p/nuttx/git/ci/9337faee31925b60a699519c4fd0506b34f0585f/

LorenzMeier commented:
Thanks Greg! The plan was to rework the patch for the upstream version (as we will ditch this branch as soon as we have the soak testing done on the upstream version) - its very nice from you that you did that step already for us, much appreciated!

nuttx commented:
The truth is that I I just get confused about which repository I am looking at.  I probably should get off of this mailing list and just monitor the newer repository.  Then I would pick up changes only after you have reviewed then and prep'ed then for upstream.

bkueng committed:
[REJECTED] arm/include/math.h: remove _PARAMS macro

_PARAMS got removed in upstream newlib:
https://github.com/bminor/newlib/commit/eea249da3bc81776246ad5163f5eb887afdd3659

davids5 commented:
@bkueng - can we re-evaluate it on the nuttx upgrade? @dagar position is we want use as much of the tool chain as safe.

bkueng commented:
That's ok. But I need this now, because I cannot compile without.
